pkg/controller/atlasproject/ipaccess_list.go

```diff
@@ -1,207 +1,133 @@
-			result := workflow.Terminate(workflow.ProjectIPNotCreatedInAtlas, fmt.Sprintf("failed to check status in Atlas: %s", err))
-			service.SetConditionFromResult(api.IPAccessListReadyType, result)
-
-			return result
-		}
-
-		if ipAccessStatus == ipAccessStatusFailed {
-			result := workflow.Terminate(workflow.ProjectIPNotCreatedInAtlas, fmt.Sprintf("configuration of %s failed in Atlas", mapToEntryValue(ipAccessList)))
-			service.SetConditionFromResult(api.IPAccessListReadyType, result)
-
-			return result
-		}
-
-		if ipAccessStatus == ipAccessStatusPending {
-			result := workflow.InProgress(workflow.ProjectIPAccessListNotActive, fmt.Sprintf("waiting Atlas to configure entry %s", mapToEntryValue(ipAccessList)))
-			service.SetConditionFromResult(api.IPAccessListReadyType, result)
-
-			return result
-		}
+	if !reflect.DeepEqual(ialInAKO.GetActives(), ialInAtlas) {
+		return i.configure(ialInAtlas, ialInAKO, isUnset)
 	}
 
-	service.SetConditionTrue(api.IPAccessListReadyType)
-
-	if len(akoProject.Spec.ProjectIPAccessList) == 0 {
-		service.UnsetCondition(api.IPAccessListReadyType)
+	if isUnset {
+		return i.unmanage()
 	}
 
-	return workflow.OK()
+	return i.progress(ialInAKO)
 }
 
-func mapToOperatorSpec(projectIPAccessList []admin.NetworkPermissionEntry) []project.IPAccessList {
-	ipAccessList := make([]project.IPAccessList, 0, len(projectIPAccessList))
-
-	for _, projectIPAccess := range projectIPAccessList {
-		deleteAfterDate := ""
-		if deleteAfterDateFromAtlas, ok := projectIPAccess.GetDeleteAfterDateOk(); ok {
-			deleteAfterDate = timeutil.FormatISO8601(*deleteAfterDateFromAtlas)
-		}
-
-		ipAccessList = append(
-			ipAccessList,
-			project.IPAccessList{
-				AwsSecurityGroup: projectIPAccess.GetAwsSecurityGroup(),
-				CIDRBlock:        projectIPAccess.GetCidrBlock(),
-				Comment:          projectIPAccess.GetComment(),
-				DeleteAfterDate:  deleteAfterDate,
-				IPAddress:        projectIPAccess.GetIpAddress(),
-			},
-		)
+// configure update Atlas with new ip access list
+func (i *ipAccessListController) configure(current, desired ipaccesslist.IPAccessEntries, isUnset bool) workflow.Result {
+	err := i.service.Add(i.ctx.Context, i.project.ID(), desired.GetActives())
+	if err != nil {
+		return i.terminate(workflow.ProjectIPNotCreatedInAtlas, err)
 	}
 
-	return ipAccessList
-}
-
-func syncIPAccessList(service *workflow.Context, projectID string, current, desired []project.IPAccessList) error {
-	currentMap := map[string]project.IPAccessList{}
-	for _, item := range current {
-		currentMap[genIPAccessListKey(item)] = item
+	for key, entry := range current {
+		if _, ok := desired[key]; !ok {
+			err = i.service.Delete(i.ctx.Context, i.project.ID(), entry)
+			if err != nil {
+				return i.terminate(workflow.ProjectIPNotCreatedInAtlas, err)
+			}
+		}
 	}
 
-	desiredMap := map[string]project.IPAccessList{}
-	for _, item := range desired {
-		desiredMap[genIPAccessListKey(item)] = item
+	if isUnset {
+		return i.unmanage()
 	}
```

Comment:
I find it confusing this `isUnset -> unmanage` is both here and in the main `reconcile` body. Can you elaborate on why is that needed?

Reply:
This ensures status conditions are removed when config is removed from the manifest